### ISSUE TITLE
Problem with flow event

### ISSUE THREAD

**Issue by Luke20000429** [AMDResearch/omnitrace#266]
I write a program which will launch kernels on 4 independent streams. However, when I profile that with omnitrace the flow event of memcpyD2H looks like this 
![image](https://user-images.githubusercontent.com/45691081/227015536-51f03525-bcd7-4d09-9b8c-187670d32ad4.png)
![image](https://user-images.githubusercontent.com/45691081/227015566-4bb1804c-55b3-48e8-93bd-8997b0276a30.png)
I thought these memcpy should be independent from each other, could someone explain why this happens?
I am running on ROCM 5.4.3 with omnitrace 1.8.0 and gfx1030.

**Comment by jrmadsen** [comment]
This looks like a bug in the correlation IDs provided by the roctracer library. It looks like roctracer is just giving OmniTrace the same correlation ID everytime so Perfetto is connecting all of them. 

Can you click on one of the entries with arrows and, in the debug annotations in the lower right of the details at the bottom, there should be a "corr_id" arg with a value -- if you right-click and there should be an option for something like "Find all arg with same value"? My guess is that will return a lot of results. 

If possible, could you verify this behavior doesn't exist in ROCm 5.3? If not and the code producing this is relatively simple, please just fork and drop it into a subdirectory in the examples folder and LMK so I can try to reproduce it.

**Comment by Luke20000429** [comment]
Thanks for your quick response, I click the `corr_id` but it seems only two entries share the same id.
For example, one of the CopyDeviceToHost has `corr_id=295`. The same id is only used by the API call.
![image](https://user-images.githubusercontent.com/45691081/227051705-a5ab4470-d494-43f8-8dc7-457855cdd477.png)
![image](https://user-images.githubusercontent.com/45691081/227051755-3a7080ee-e801-4d38-9ec0-9aadec84d215.png)

I didn't install ROCm 5.3, so I might switch to another workstation. If that doesn't work, I will add a simple demo to my forked repo.

**Comment by jrmadsen** [comment]
> I didn't install ROCm 5.3, so I might switch to another workstation. If that doesn't work, I will add a simple demo to my forked repo.

Ah no need, I located the problem. I was accidentally using the internal correlation id for critical tracing (which intentionally makes the connections you are seeing) instead of using the roctracer correlation id. I didn't realize that was the case bc I did a very poor job naming the variables: the former variable name was `_cid` and the latter variable name was `_corr_id`. I'll get that fixed and generate a v1.9.1 release.
